### PR TITLE
Fix black screen on initial render on macOS

### PR DESCRIPTION
**Pull request by raccube**
Credit for fix from https://github.com/raysan5/raylib/pull/655

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -32,6 +32,13 @@
 
 #include "algo.h"
 
+#if defined(__APPLE__)
+    #include <unistd.h>             // Required for: usleep()
```
Comment (coderobe) [inline src/main.cpp]: usleep is never used

File: meson.build [hunk 2]
```diff
@@ -53,6 +53,10 @@ dependencies = [
   dependency('glew'),
 ]
 
+if host_machine.system() == 'darwin'
+  dependencies += dependency('Foundation')
```
Comment (coderobe) [inline meson.build:57]: why is this required?

File: src/main.cpp [hunk 3]
```diff
@@ -116,6 +123,9 @@ void fill_targets(){
 }
 
 void render(){
+  // macOS Mojave rerender fix
+  int windowNeedsUpdating = 2;
```
Comment (coderobe) [inline src/main.cpp]: did you mean for this to be `static`? otherwise this will never actually decrease as `windowNeedsUpdating--` is called only once after its usage in `render()`

Comment (coderobe) [reply src/main.cpp]: also, please use a more... descriptive... name than this. `render()` is pretty big.

File: src/main.cpp [hunk 4]
```diff
@@ -207,6 +217,15 @@ void render(){
     glClearColor(0, 0, 0, 0);
     nk_glfw3_render(NK_ANTI_ALIASING_OFF, MAX_VERTEX_BUFFER, MAX_ELEMENT_BUFFER);
     glfwSwapBuffers(win);
+#if __APPLE__
+    // workaround for missing/erroneous initial rendering on macOS
+    if (windowNeedsUpdating) {
+        // Desugared version of Objective C: [glfwGetNSGLContext(window) update]
+        ((id (*)(id, SEL))objc_msgSend)(glfwGetNSGLContext(win),
```
Comment (coderobe) [inline src/main.cpp]: that's a pretty disgusting cast, is that really necessary?

**Review by coderobe — CHANGES_REQUESTED** [review]
i'm pretty sure this will indefinitely dispatch update events, please test your code..

File: src/main.cpp (hunk 1 above)
Comment (raccube) [reply src/main.cpp]: I thought I had removed that but it turns out this was after committing.

File: meson.build (hunk 2 above)
Comment (raccube) [reply meson.build:57]: Contains `objc_msgSend()`

Comment (coderobe) [reply meson.build:57]: i see
